src/mouse.cpp

```diff
@@ -1478,10 +1553,13 @@ void GraphicsWindow::EditControlDone(const std::string &s) {
 
             default:
                 // These are always positive, and they get the units conversion.
-                c->valA = fabs(SS.ExprToMm(e));
+                // Use ExprToMm since this unparameterized expression will simplify immediately, and the dimension transformation is irrelevant
+                c->valA = fabs(SS.NonConstraintExprToMm(e));
                 break;
         }
+        SK.GetGroup(c->group)->dofCheckOk = false; // if an named param was used in the constraint, the DoF may have changed
         SS.MarkGroupDirty(c->group);
+        SS.TW.HideEditControl();
```

Comment:
Why did you have to add this? (I have not tried to remove it and debug what happens).